### PR TITLE
Update SQLAlchemy

### PR DESCRIPTION
**Pull request by ben-freenome**
fix code and tests to work with newer sqlalchemy - works with 1.1 and 1.2. 1.3 has problems, but that's in beta so I'll punt on that.

### CODE REVIEW

File: flask_restless/search.py [hunk 1]
```diff
@@ -465,7 +466,10 @@ def create_query(session, model, search_params, _ignore_order_by=False):
                         relation_model = relation.mapper.class_
                         field = getattr(relation_model, field_name_in_relation)
                         direction = getattr(field, val.direction)
-                        query = query.join(relation_model)
+                        if relation_model not in relation_models:
```
Comment (pnachum) [inline flask_restless/search.py]: What's this for? I thought `join` already handles this?

File: tests/test_search.py [hunk 2]
```diff
@@ -164,17 +160,16 @@ def test_order_by_and_filter_on_the_same_relation(self):
...
+        assert results[0].owner.name == 'Lucy'
+        assert results[1].owner.name == 'Mary'
 
     def test_order_by_two_different_fields_of_the_same_relation(self):
-
         # Given
-        computer_for_mary = self.Computer(name=u'1st', vendor=u'vendor')
-        mary = self.session.query(self.Person).filter_by(name=u'Mary').first()
+        computer_for_mary = self.Computer(name='1st', vendor='vendor')
+        mary = self.session.query(self.Person).filter(self.Person.name == 'Mary').first()
```
Comment (pnachum) [inline tests/test_search.py]: what's this change for?

Comment (ben-freenome) [reply tests/test_search.py]: whoops, I'll revert.

File: flask_restless/search.py (hunk 1 above)
Comment (ben-freenome) [reply flask_restless/search.py]: it doesn't seem to always; without this check we get two identical joins in some tests in sqlalchemy 1.2

**Pull request merged**